specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/webapplicationfirewall.json

```diff
@@ -534,6 +567,115 @@
+    "properties": {
+      "ruleSets": {
+        "description": "List of rules",
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/ManagedRuleSet"
+        }
+      }
+    }
+  },
+  "ManagedRuleSet": {
+    "type": "object",
+    "description": "Base class for all types of ManagedRuleSet.",
+    "x-ms-discriminator-value": "Unknown",
+    "properties": {
+      "ruleSetVersion": {
+        "description": "defines version of the rule set",
+        "type": "string"
+      },
+      "ruleSetType": {
+        "description": "RuleSetType - AzureManagedRuleSet or OWASP RuleSets.",
+        "type": "string"
+      }
+    },
+    "required": [
+      "ruleSetType"
+    ],
+    "discriminator": "ruleSetType"
+  },
+  "OWASPRuleSet": {
+    "type": "object",
+    "description": "Describes azure managed provider.",
+    "x-ms-discriminator-value": "OWASPRuleSet",
+    "properties": {
+      "ruleSetConfiguration": {
+        "$ref": "#/definitions/OWASPRuleSetConfiguration",
+        "description": "OWASP rule set configuration."
+      }
+    },
+    "allOf": [
+      {
+        "$ref": "#/definitions/ManagedRuleSet"
+      }
+    ]
+  },
+  "OWASPRuleSetConfiguration": {
+    "properties": {
+      "disabledRuleGroups": {
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/OWASPRuleSetDisabledRuleGroup"
+        },
+        "description": "The disabled rule groups."
+      },
+      "exclusions": {
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/OWASPRuleSetExclusion"
+        },
+        "description": "The exclusion list."
+      }
+    },
+    "description": "OWASP rule set configuration."
+  },
+  "OWASPRuleSetDisabledRuleGroup": {
+    "properties": {
+      "ruleGroupName": {
+        "type": "string",
+        "description": "The name of the rule group that will be disabled."
+      },
+      "rules": {
+        "type": "array",
+        "items": {
+          "type": "integer",
+          "format": "int32",
+          "x-nullable": false
+        },
+        "description": "The list of rules that will be disabled. If null, all rules of the rule group will be disabled."
+      }
+    },
+    "required": [
+      "ruleGroupName"
+    ],
+    "description": "Allows to disable rules within a rule group or an entire rule group."
+  },
+  "OWASPRuleSetExclusion": {
+    "properties": {
+      "matchVariable": {
+        "type": "string",
+        "description": "The variable to be excluded."
+      },
+      "selectorMatchOperator": {
+        "type": "string",
+        "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to."
+      },
+      "selector": {
+        "type": "string",
+        "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
+      }
+    },
+    "required": [
```

Comment:
Please check are all of these items required?
Say, for [ApplicationGatewayFirewallExclusion](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json#L2334) I was able to create it using matchVariable only (calling API directly).
I.e. I omitted `selectorMatchOperator` and `selector` both and it still worked well.
Is it also the case there?

Reply:
Only matchVariable is required. There can be a case with selector and selectorMatchOperator as null. I will address the same.